### PR TITLE
Time Changing Vaccine Efficacy Changes

### PR DESCRIPTION
**Pull request by OJWatson**
Start of a PR to adapt to new nimue version (0.1.18), which has time changing vaccine efficacies.

This is a draft as it is half way there for implementing:

1. Methods for parameterising the vaccine efficacies over time, which now need to be 3 dimensional arrays of [nAge, nVaccine, timesteps]
2. Updated Reff calculation as this now requires using the time changing vaccine properties
3. Taking inputs from the UI on future vaccine doses to be given out and working out what the weighted vaccine efficacies should be in the future. 

Most of these changes have been written up in `R` so far but I am very out of my depth translating this to `js`. Priority is 1 and 2. 3 being longer term is okay

### CODE REVIEW

File: src/parameters.js [hunk 1]
```diff
@@ -90,6 +92,25 @@ export const createParameters = (
       this.infectionEfficacy[4] = vaccinatedInfectedEff;
       return this;
     },
+    withVaccineEfficacy: function(diseaseEfficacy, infectionEfficacy) {
```
Comment (OJWatson) [inline src/parameters.js]: this shouldn;t been duplicated.my bad

File: R/reff.R [hunk 2]
```diff
@@ -1,51 +1,81 @@
...
     out$parameters$contact_matrix_set[[1]],
     out$parameters$population
   )
-
+  
   # these parameters are found in pars_0.json that is imported in index.js
   dur_ICase <- out$parameters$dur_ICase
   dur_IMild <- out$parameters$dur_IMild
-  prob_hosp <- out$odin_parameters$prob_hosp
+  rel_infectiousness <- out$odin_parameters$rel_infectiousness_vaccinated
```
Comment (giovannic) [inline R/reff.R:12]: We can remove this?

**Review by giovannic — COMMENTED** [review]
The Reff function and vaccine parameterisation appears to work from the unittests.

...it probably isn't super fast...

Are you up for polishing up the documentation and releasing?

I haven't done an e2e test for varying efficacy. I can leave that to you if you want?

**Comment by giovannic** [comment]
Oh, there's a weird babel issue. Will look into it

File: R/reff.R (hunk 2 above)
Comment (OJWatson) [reply R/reff.R:12]: Yup we can. sorry for the confusion

**Comment by OJWatson** [comment]
> The Reff function and vaccine parameterisation appears to work from the unittests.
> 
> ...it probably isn't super fast...
> 
> Are you up for polishing up the documentation and releasing?
> 
> I haven't done an e2e test for varying efficacy. I can leave that to you if you want?

Happy to write the documentation etc for this as will be chatting to Andy to help explain how to plumb it all together. i can then write a test to use one of the current model fits to make sure it is working right.

**Comment by giovannic** [comment]
Ok, sorry for the late commit...

[Babel is breaking](https://github.com/mapbox/mapbox-gl-js/issues/3422) and I don't know why or how to fix it. So I've removed it.

It's not essential that we spend time on maintaining Babel. Babel just makes sure that our javascript bundle works on all recent browsers. But I'm sure clients can figure out a Babel configuration that works for them and run the bundle through it themselves.

I've updated the e2e/test.js to use the new vaccine efficacy methods.

**Pull request merged**